src/arrayMethodReduce.js

```diff
@@ -4,8 +4,20 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, startValue) {
+    let prev = startValue;
+    let start = 0;
+
+    if (arguments.length < 2) {
+      start = 1;
+      prev = this[0];
+    }
+
+    for (let i = start; i < this.length; i++) {
+      prev = callback(prev, this[i], i, this);
+    }
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )